app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -701,25 +715,34 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
 
         // Detach any existing mediator before changing tab contents and attaching a new mediator
         tabLayoutMediator?.detach()
+        currentTabs.clear()
 
         tabAdapter.tabs = tabs
         tabAdapter.notifyItemRangeChanged(0, tabs.size)
 
         tabLayoutMediator = TabLayoutMediator(activeTabLayout, binding.viewPager, true) {
                 tab: TabLayout.Tab, position: Int ->
             tab.icon = AppCompatResources.getDrawable(this@MainActivity, tabs[position].icon)
-            tab.contentDescription = when (tabs[position].id) {
+            tab.contentDescription = when (tabs[position].kind) {
                 LIST -> tabs[position].arguments[1]
                 else -> getString(tabs[position].text)
             }
+            if (tabs[position].kind == DIRECT) {
+                tab.orCreateBadge
+                tab.badge?.isVisible = false
+
+                // TODO! the badge state must also be stored ("there are still new messages even if you restart Tusky")
```

Comment:
(A number would probably be a bit over the top. It's only a dot.)

What kind of an "UI affordance" do you have in mind?
It should of course be something automatic. Like with all notifications: disappear when opened / read.

Reply:
A dot is fine.

The UI afforance is either an icon or menu item on the existing per-notification menu, possibly both.

Note that not all types of notification currently have a menu or row of icons (e.g., "X boosted your post", "X favourited your post" notifications do not). So this will need to be added.

In this case I don't think marking a notification as read should be automatic, because the API provides no mechanism to mark a notification as unread again. So marking a notification as read should require explicit, positive confirmation from the user.

The user should be able to mark multiple notifications in the list to mark them as read. RecyclerView supports doing this, triggered by the user long-pressing an item in the list. https://www.kodeco.com/29024188-recyclerview-selection-library-tutorial-for-android-adding-new-actions is a decent tutorial on how to do that.